### PR TITLE
feat(kv): implement labels generically on kv

### LINKED ISSUE #11724: kv: review port of labels (compare and contrast with bolt)

### PR DESCRIPTION
**Pull request by desa**
Closes #11724 

_Briefly describe your proposed changes:_
This PR migrates labels to kv.

  - [ ] Rebased/mergeable
  - [ ] Tests pass
  - [ ] http/swagger.yml updated (if modified Go structs or API)
  - [ ] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: kv/label.go [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	"github.com/influxdata/influxdb"
+)
+
+var (
+	labelBucket        = []byte("labelsv1")
```
Comment (leodido) [inline kv/label.go]: Not `"labels/v1"` like in other refactored services?

Comment (desa) [reply kv/label.go]: https://github.com/influxdata/influxdb/pull/11801#discussion_r255796007

File: kv/label.go [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+
+	rid, err := m.ResourceID.Encode()
+	if err != nil {
+		return nil, &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Err:  err,
+		}
+	}
+
+	key := make([]byte, len(rid)+len(lid))
```
Comment (leodido) [inline kv/label.go]: Maybe we can use constants (ie., `influxdb.IDLength`) here rather then `len()`.

File: kv/label.go (hunk 1 above)
Comment (leodido) [reply kv/label.go]: Gotcha!

**Pull request merged**